owncloudApp/src/main/java/com/owncloud/android/providers/DocumentsStorageProvider.kt

```diff
@@ -190,18 +193,50 @@ class DocumentsStorageProvider : DocumentsProvider() {
         val file = currentStorageManager?.getFileById(docId) ?: throw FileNotFoundException("File $docId not found")
         Log_OC.d(TAG, "Trying to rename ${file.fileName} to $displayName")
 
-        val renameFileOperation = RenameFileOperation(file.remotePath, displayName)
-        val result = renameFileOperation.execute(currentStorageManager, context)
+        RenameFileOperation(file.remotePath, displayName).apply {
+            execute(currentStorageManager, context).also { checkOperationResult(it, file) }
+        }
+
+        return null
+    }
+
+    override fun deleteDocument(documentId: String) {
+        val docId = documentId.toLong()
+
+        updateCurrentStorageManagerIfNeeded(docId)
+
+        val file = currentStorageManager?.getFileById(docId) ?: throw FileNotFoundException("File $docId not found")
+        Log_OC.d(TAG, "Trying to delete ${file.fileName} with id ${file.fileId}")
+
+        RemoveFileOperation(file.remotePath, false).apply {
```

Comment:
Wow, taking advantage of Kotlin power, just a couple of lines to execute the operation and check the result, love it 👏